DelvUI/Config/Tree/ConfigNode.cs

```diff
@@ -1038,6 +1035,30 @@ public CategoryField(FieldInfo mainField, PluginConfigObject configObject, strin
 
         public void AddChild(int position, FieldInfo field) { Children.Add(position, field); }
 
+        public void AddSeparator(FieldInfo field)
+        {
+            foreach (object attribute in field.GetCustomAttributes(true))
+            {
+                if (attribute is CheckboxAttribute { friendlyName: "Enabled", separator: true } && ID != null) { }
+                else if (attribute is DragFloatAttribute { separator: true }) {}
+                else if (attribute is DragIntAttribute { separator: true }) {}
+                else if (attribute is DragFloat2Attribute { separator: true }) {}
+                else if (attribute is DragInt2Attribute { separator: true }) {}
+                else if (attribute is InputTextAttribute { separator: true }) {}
+                else if (attribute is ColorEdit4Attribute { separator: true }) {}
+                else if (attribute is ComboAttribute { separator: true }) {}
+                else if (attribute is DragDropHorizontalAttribute { separator: true }) {}
+                else if (attribute is DynamicList { separator: true }) { }
+                else { continue; }

```

Comment:
If the only special case is CheckboxAttribute you should do something like this:
```
if (attribute is CheckboxAttribute { friendlyName: "Enabled", separator: true } && ID != null) { }
else if (attribute is ConfigAttribute { separator: true }) {}
else { continue; }
```
Since the property is on the base class thats all you need to check for.

Reply:
hmm ok nice, i was struggling to figure this solution